frontend/components/vault/VaultFlow.vue

```diff
@@ -0,0 +1,170 @@
+import VaultsText from './VaultsText.vue';
+import Vault from './Vault.vue';
+import VaultLiquidationTransactionFlow from './VaultLiquidationTransactionFlow.vue';
+import SplitLayout from '~/components/layout/SplitLayout.vue';
+import LandingBlock from '~/components/layout/LandingBlock.vue';
+
+export default Vue.extend({
+    components: {
+        VaultLiquidationTransactionFlow,
+        Vault,
+        VaultsText,
+        SplitLayout,
+        LandingBlock,
+    },
+    props: {
+        vaultTransactions: {
+            type: Array as Vue.PropType<VaultTransaction[]>,
+            default: () => [],
+        },
+        areVaultsFetching: {
+            type: Boolean,
+            default: false,
+        },
+        vaultsError: {
+            type: String,
+            default: null,
+        },
+        selectedVaultId: {
+            type: String,
+            default: null,
+        },
+        isConnectingWallet: {
+            type: Boolean,
+            default: false,
+        },
+        isRefreshingLimits: {
+            type: Boolean,
+            default: false,
+        },
+        isLiquidating: {
+            type: Boolean,
+            default: false,
+        },
+        walletAddress: {
+            type: String,
+            default: null,
+        },
+        isExplanationsShown: {
+            type: Boolean,
+            default: true,
+        },
+        lastUpdated: {
+            type: Date,
+            default: null,
+        },
+        network: {
+            type: String,
+            default: 'mainnet',
+        },
+    },
+    data: () => ({
+        step: 0,
+    }),
+    computed: {
+        selectedVaultTransaction(): VaultTransaction | null {
+            return (
+                this.vaultTransactions.find(
+                    vaultTransaction => vaultTransaction.id.toString() === this.selectedVaultId
+                ) || null
+            );
+        },
+        isStagingEnvironment(): boolean {
+            return !!process.env.STAGING_BANNER_URL;
+        },
+    },
+    watch: {
+        selectedVaultId: {
+            immediate: true,
+            handler(newSelectedVaultId) {
+                if (!newSelectedVaultId) {
+                    this.step = 0;
+                } else {
+                    this.step = 1;
+                }
+            },
+        },
+        step: {
+            immediate: true,
+            handler(newStep) {
+                if (newStep === 0) {
+                    this.$emit('update:selectedVaultId', null);
+                }
+            },
+        },
+    },
+    methods: {
+        explanationsTrigger(event: boolean): void {
+            if (event === true && this.$refs.mainText) {
+                (this.$refs.mainText as Vue).$el.scrollIntoView({ block: 'start', behavior: 'smooth' });
+            }
+            this.$emit('update:isExplanationsShown', event);
+        },
+        liquidate() {
+            this.step = 2;
+        },
```

Comment:
The name of this method is misleading (as well as the emitted event name) Let's find a different name for it (eg `participate`/`openLiquidationTransaction`)